### PR TITLE
Add getDeviceDetails action

### PR DESCRIPTION
**Pull request by bachya**
**Checks**

- [x] I've checked there are no linting errors.
- [x] I've checked the code is still at 100% test coverage.

**Added Actions (if relevant)**

- `Get Device Details`

**Are you happy to be listed as a contributor on [Shortcuts.fun](https://shortcuts.fun)?**

Yes

**Any other information / comments**

N/A

### CODE REVIEW

**Comment by bachya** [comment]
@joshfarrant One thing I don't see any examples of how we handle "Ask When Run" parameter values. Any thoughts?

**Comment by bachya** [comment]
Great! Thanks so much, @xAlien95!

**Comment by bachya** [comment]
@xAlien95, quick follow-up: what would I do in situations where “Ask When Run” is part of an option set?

Image

**Comment by xAlien95** [comment]
> ... what would I do in situations where “Ask When Run” is part of an option set?

@bachya, you have to add `WFSerialization` as possible type for `WFDeviceDetail`:

```ts
const getDeviceDetails = (
  options: {
    /** The particular detail to retrieve */
    detail?: WFDeviceDetail | WFSerialization,
  },
): WFWorkflowAction => {
  const {
    detail = 'Device Name',
  } = options;

  return {
    WFWorkflowActionIdentifier: 'is.workflow.actions.getdevicedetails',
    WFWorkflowActionParameters: {
      WFDeviceDetail: detail,
    },
  };
};
```

Take a look at [`actions/text.ts`](https://github.com/joshfarrant/shortcuts-js/blob/master/src/actions/text.ts#L18) as an example.

**Comment by bachya** [comment]
@xAlien95 I believe I added support correctly, but, inexperienced as I am, I can't seem to test it successfully (i.e., I can't establish what code produces a shortcut that actually uses "Ask When Run" in this case). Please correct me where I'm wrong!

File: src/actions/getDeviceDetails.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * ```js
+ * getDeviceDetails({
+ *   'detail': 'Device Name',
+ * });
+ * ```
+ */
+const getDeviceDetails = (
+  options: {
+    /** The particular detail to retrieve */
+    detail?: WFDeviceDetail,
```
Comment (xAlien95) [inline src/actions/getDeviceDetails.ts]: Could you change this line to the following?
```ts
detail?: WFSerialization | WFDeviceDetail,
```

File: src/interfaces/WF/WFDeviceDetail.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+type WFDeviceDetail = (
+  'Device Name'
+  | 'Device Model'
+  | 'System Version'
+  | 'Screen Width'
+  | 'Screen Height'
+  | 'Current Volume'
+  | 'Current Brightness'
+  | WFSerialization
```
Comment (xAlien95) [inline src/interfaces/WF/WFDeviceDetail.ts]: Could you remove this line?

File: src/interfaces/WF/WFWorkflowActionParameters.ts [hunk 3]
```diff
@@ -29,6 +30,7 @@ interface WFWorkflowActionParameters {
   WFControlFlowMode?: number;
   WFCountType?: WFCountType;
   WFDelayTime?: number;
+  WFDeviceDetail?: WFDeviceDetail;
```
Comment (xAlien95) [inline src/interfaces/WF/WFWorkflowActionParameters.ts]: Could you change this line to the following?
```ts
WFDeviceDetail?: WFSerialization | WFDeviceDetail;
```

File: src/actions/getDeviceDetails.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import WFDeviceDetail from '../interfaces/WF/WFDeviceDetail';
+import WFSerialization from '../interfaces/WF/WFSerialization';
+import WFWorkflowAction from '../interfaces/WF/WFWorkflowAction';
+
+/**
+ * Get Device Details Action. Returns a particular property of the device.
+ *
+ * ```js
+ * getDeviceDetails({
+ *   'detail': 'Device Name',
```
Comment (xAlien95) [inline src/actions/getDeviceDetails.ts]: Could you change this line with the following?
```ts
detail: 'Device Name',
```

**Comment by xAlien95** [comment]
@bachya, thanks for looking into it. I added some comments to let you separate `WFDeviceDetail` type from the `WFSerialization` (`askWhenRun` variable).

> I can't establish what code produces a shortcut that actually uses "Ask When Run" in this case

At the moment `askWhenRun` global variable is based on the `Attachment` type, in the next few days I'll present the a pull request in which variables are based on `WFSerialization`.

If you want to test now your action with `askWhenRun`, you have to write the following:
```js
getDeviceDetails({
  detail: {
    Value: askWhenRun,
    WFSerializationType: 'WFTextTokenAttachment',
  },
})
```

In the future, you'll just have to write:
```js
getDeviceDetails({
  detail: askWhenRun,
})
```

**Comment by joshfarrant** [comment]
Thanks both, looks good 👍

File: src/actions/getDeviceDetails.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { withActionOutput } from '../utils';
+
+import WFDeviceDetail from '../interfaces/WF/WFDeviceDetail';
+import WFSerialization from '../interfaces/WF/WFSerialization';
+import WFWorkflowAction from '../interfaces/WF/WFWorkflowAction';
+
+/**
+ * Get Device Details Action. Returns a particular property of the device.
```
Comment (joshfarrant) [inline src/actions/getDeviceDetails.ts]: Could this be changed to _Get Device Details Action. Gets information about the current device._ please? Just to match the description in the Shortcuts app.

**Pull request merged**